### PR TITLE
added route to update marketplace user password

### PR DESCRIPTION
**Pull request by jonmchu**
### What does this PR do?
Edited Marketplace_user_controller to provide route to change user password. 

### Status
READY

### Migrations
YES 

### Description of Testing Done
Tested on react native apps

CC: @nhiquach

### CODE REVIEW

File: db/migrate/20171110025825_add_password_reset_to_marketplace_user.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddPasswordResetToMarketplaceUser < ActiveRecord::Migration
+  def change
+    add_column :marketplace_users, :password_reset, :boolean
```
Comment (amillman) [inline db/migrate/20171110025825_add_password_reset_to_marketplace_user.rb]: can we rename it to like `:requires_password_reset`? the naming is a bit confusing at first

Comment (amillman) [reply db/migrate/20171110025825_add_password_reset_to_marketplace_user.rb]: or `:has_reset_password`

File: app/controllers/api/marketplace/marketplace_users_controller.rb [hunk 2]
```diff
@@ -33,6 +33,20 @@ def listings
     render json: marketplace_listings, root: false
   end
 
+  def update_password
+    authenticate_user_from_token!
+    marketplace_user = MarketplaceUser.find(params[:id])
+    if marketplace_user.valid_password?(params[:current_password])
+      if marketplace_user.update(update_password_params)
+        render_json_message(:ok, message: 'User successfully updated password!')
```
Comment (nhiquach) [inline app/controllers/api/marketplace/marketplace_users_controller.rb]: We can use error class (error_response) once my changes are merged

File: app/controllers/api/marketplace/marketplace_users_controller.rb [hunk 3]
```diff
@@ -33,6 +33,20 @@ def listings
...
+      else
+        render_json_message(:forbidden, errors: marketplace_user.errors.full_messages)
```
Comment (nhiquach) [inline app/controllers/api/marketplace/marketplace_users_controller.rb]: We can use error class (error_response) once my changes are merged

File: app/serializers/marketplace_user_serializer.rb [hunk 4]
```diff
@@ -33,5 +33,6 @@ class MarketplaceUserSerializer < ActiveModel::Serializer
              :marketplace_region_id,
              :company_name,
              :first_name,
-             :last_name
+             :last_name,
+             :password_reset
```
Comment (nhiquach) [inline app/serializers/marketplace_user_serializer.rb]: rename (see millman's comment)

**Pull request merged**